client/engine/store/persiststore.go

```diff
@@ -0,0 +1,446 @@
+package store
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel"
+	"github.com/statechannels/go-nitro/channel/consensus_channel"
+	"github.com/statechannels/go-nitro/crypto"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/protocols/directdefund"
+	"github.com/statechannels/go-nitro/protocols/directfund"
+	"github.com/statechannels/go-nitro/protocols/virtualdefund"
+	"github.com/statechannels/go-nitro/protocols/virtualfund"
+	"github.com/statechannels/go-nitro/types"
+	"github.com/tidwall/buntdb"
+)
+
+type PersistStore struct {
+	objectives         *buntdb.DB
+	channels           *buntdb.DB
+	consensusChannels  *buntdb.DB
+	channelToObjective *buntdb.DB
+
+	key     string // the signing key of the store's engine
+	address string // the (Ethereum) address associated to the signing key
+}
+
+func (ms *PersistStore) checkError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
```

Comment:
Nit: maybe we should call this `panicOnError` or something?

Also, we want to avoid panicking in production code and generally we should prefer handling errors. Perhaps it could be justified here since this code runs at startup?

Reply:
> Also, we want to avoid panicking in production code and generally we should prefer handling errors. Perhaps it could be justified here since this code runs at startup?

One thing to note is that we could use [recover](https://gobyexample.com/recover) to capture these panics and prevent them from escaping go-nitro ([It looks like some parts of the go standard library do that](https://github.com/golang/go/blob/07a25a9f6e46e83a6335d43b0174dfb250bca960/src/fmt/print.go#L587)). 

Also for debugging purposes `panics` are quite powerful, they immediately stop things at the site of the problem and require almost no additional error handling logic. Obviously this isn't ideal for production code but I'd prefer code that `panic`s on an error instead of ignoring it (which has been the cause of some painful bugs).

I think this could be fertile ground for an ADR or RFC? Some parts of the standard library do [panic](https://github.com/golang/go/blob/2d84fbbad8e939ccf82be1b9addf26f73f39f2dc/src/bufio/scan.go#L167) so it could be worthwhile considering the use of `panic`s and basic error handling in general.